### ISSUE TITLE
Modify time unit on zoom

### ISSUE THREAD

**Issue by Techn1x** [chartjs/chartjs-plugin-zoom#179]
On a time scale chart, as the chart is zoomed in, the time gets more accurate, but the time axis remains as the same time display unit as when the chart was first loaded. Ideally when zooming the chart would be able to recognise the new chart limits, re-calculate the unit to use (unless it's explicitly specified already by the options), and use that to pick a display format.

https://www.chartjs.org/docs/latest/axes/cartesian/time.html#time-units

```javascript
options: {
  scales: {
    xAxes: [{
      type: 'time',
      time: {
        unit: 'month', // This should change when zoomed
        displayFormats: {
          minute: 'h:mm a',
          month: 'MMM YYYY'
        }
      }
    }]
  }
}
```

**Comment by Techn1x** [comment]
If you need this functionality now, this could potentially be done manually in onZoom() if you have access to the chart instance to calculate the new min/max bounds yourself (note my PR that gives this access https://github.com/chartjs/chartjs-plugin-zoom/pull/178 ) and then set the new chart options then update the chart

**Comment by rdgout** [comment]
> If you need this functionality now, this could potentially be done manually in onZoom() if you have access to the chart instance to calculate the new min/max bounds yourself (note my PR that gives this access #178 ) and then set the new chart options then update the chart

Hi,

I'm trying to accomplish exactly what this issue asks for.
However, I'm not sure how you would determine these bounds based on the chart instance. 
Unless I'm missing it, there is no way to determine how far we are zoomed in. Even the amount of ticks is not accurate when checking the scale.

**Comment by benmccann** [comment]
You can check the scale min and max

**Comment by kurkle** [comment]
If you don't specify the unit in time scale options, it will determine the unit automatically.

This can be seen in action in this sample: https://www.chartjs.org/chartjs-plugin-zoom/samples/time.html

**Comment by kurkle** [comment]
Closing, works as expected

**Comment by ibtisamarif831** [comment]
My chart had initial unit set to year on the first render and I had manually defined the unit. On zoom complete, I just set unit to false.

Because I only needed my currently defined unit on first render but on zoom, chartjs should automatically define the unit based on zoom level.

if I have 5 years of data on first load, i will see what i need to but on subsequent zooms, just reset the unit on chart object and from there, everything can be automatically handled. hope it helps someone.
